BaikalDataGenerator/Source/render.cpp

```diff
@@ -0,0 +1,437 @@
+
+    for (const auto output_info: outputs_collection)
+    {
+        m_outputs.push_back(m_factory->CreateOutput(output_width, output_height));
+        m_renderer->SetOutput(output_info.type, m_outputs.back().get());
+    }
+
+    m_scene = (file_name.empty() || path.empty()) ? Baikal::SceneIo::LoadScene("sphere+plane.test", "") :
+                                                    Baikal::SceneIo::LoadScene(file_name, path);
+}
+
+void Render::LoadMaterialXml(const std::string &path, const std::string &file_name)
+{
+    std::stringstream ss;
+    ss << path << "/" << file_name;
+
+    // Check it we have material remapping
+    std::ifstream in_materials(ss.str());
+    std::ifstream in_mapping(ss.str());
+
+    if (in_materials && in_mapping)
+    {
+        in_materials.close();
+        in_mapping.close();
+
+        auto material_io = Baikal::MaterialIo::CreateMaterialIoXML();
+        auto mats = material_io->LoadMaterials(ss.str());
+        auto mapping = material_io->LoadMaterialMapping(ss.str());
+
+        material_io->ReplaceSceneMaterials(*m_scene, *mats, mapping);
+    }
+}
+
+void Render::LoadCameraXml(const std::string &path, const std::string &file_name)
+{
+    std::stringstream ss;
+
+    ss << path << "/" << file_name;
+
+    tinyxml2::XMLDocument doc;
+    doc.LoadFile(ss.str().c_str());
+    auto root = doc.FirstChildElement("cam_list");
+
+    if (!root)
+    {
+        throw std::runtime_error("Render::LoadCameraXml(...):"
+                                 "Failed to open lights set file.");
+    }
+
+    tinyxml2::XMLElement* elem = root->FirstChildElement("camera");
+
+    m_camera_states.clear();
+
+    while (elem)
+    {
+        CameraInfo cam_info;
+
+        // eye
+        cam_info.pos.x = elem->FloatAttribute("cpx");
+        cam_info.pos.y = elem->FloatAttribute("cpy");
+        cam_info.pos.z = elem->FloatAttribute("cpz");
+
+        // center
+        cam_info.at.x = elem->FloatAttribute("tpx");
+        cam_info.at.y = elem->FloatAttribute("tpy");
+        cam_info.at.z = elem->FloatAttribute("tpz");
+
+        // up
+        cam_info.up.x = elem->FloatAttribute("upx");
+        cam_info.up.y = elem->FloatAttribute("upy");
+        cam_info.up.z = elem->FloatAttribute("upz");
+
+        //other values
+        cam_info.focal_length = elem->FloatAttribute("focal_length");
+        cam_info.focus_distance = elem->FloatAttribute("focus_dist");
+        cam_info.aperture = elem->FloatAttribute("aperture");
+
+        m_camera_states.push_back(cam_info);
+        elem = elem->NextSiblingElement("camera");
+    }
+}
+
+void Render::LoadLightXml(const std::string &path, const std::string &file_name)
+{
+    std::stringstream ss;
+
+    ss << path << "/" << file_name;
+
+    tinyxml2::XMLDocument doc;
+    doc.LoadFile(ss.str().c_str());
+    auto root = doc.FirstChildElement("light_list");
+
+    if (!root)
+    {
+        throw std::runtime_error("Render::LoadLightXml(...):"
```

Comment:
Add the file name to the message.
Also, there is `__func__` (char[]) that should contain `Render::LoadLightXml`.

Reply:
done